### PR TITLE
profile.CheckValid should return an error if location has a line with empty function

### PR DESCRIPTION
**Pull request by Labutin**
CheckValid should check that all lines in locations have non nil function pointer.

### CODE REVIEW

File: profile/profile.go [hunk 1]
```diff
@@ -402,6 +402,8 @@ func (p *Profile) CheckValid() error {
 				if f.ID == 0 || functions[f.ID] != f {
 					return fmt.Errorf("inconsistent function %p: %d", f, f.ID)
 				}
+			} else {
+				return fmt.Errorf("profile has nil function in location line")
```
Comment (aalexand) [inline profile/profile.go:406]: Can we include any context in the error message that will help triage the error?

Comment (aalexand) [inline profile/profile.go:406]: Do the nil check first and unindent the main path.

Comment (Labutin) [reply profile/profile.go]: Done

Comment (Labutin) [reply profile/profile.go]: Done

**Pull request merged**